habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PyenvAndPoetrySetup.java

```diff
@@ -91,13 +99,18 @@ class PyenvAndPoetrySetup {
      * @param log                            the logger to use for output
      */
     public PyenvAndPoetrySetup(String pythonVersion, boolean usePyenv, File patchInstallScript,
-                               File baseDir, boolean rewriteLocalPathDepsInArchives, Log log) {
+                               File baseDir, boolean rewriteLocalPathDepsInArchives,
+                               String username, String password, String pypiRepoId, String pypiRepoUrl, Log log) {
         this.pythonVersion = pythonVersion;
         this.usePyenv = usePyenv;
         this.patchInstallScript = patchInstallScript;
         this.baseDir = baseDir;
         this.rewriteLocalPathDepsInArchives = rewriteLocalPathDepsInArchives;
         this.log = log;
+        this.username = username;
+        this.password = password;
+        this.pypiRepoId = pypiRepoId;
```

Comment:
A: If it simplifies the code, it would make sense to wrap the repo credentials and ID in an object.

Reply:
For now I will leave it because the object could sometimes use the repo url and sometimes not, but this could be good for the future. I'm just not sure it simplifies the code right now.